lib/fuzion/std/fileio.fz

```diff
@@ -137,19 +137,25 @@ public fileio is
                 # the (relative or absolute) file name, using platform specific path separators
                 path string) outcome unit is
     arr := path.utf8.asArray
-    if delete arr.internalArray.data arr.length
-      unit
-    else
-      error "an error occurred while performing the delete operation on the following file/dir: \"$path\""
+    match exists path
+      b bool =>
+        if b = TRUE
+          if delete arr.internalArray.data unit
+            unit
+          else
```

Comment:
instead of 
```
if f a b c
  unit
else
  error
```
you should better write
```
if ! f a b c
  error
```
or
```
if !(f a b c)
  error
```